### DISCUSSION TITLE
Node coordinates

### DISCUSSION THREAD

**Discussion by wangjx520** [jtecklenburg/voromesh#2] [General]
This is a very good work. I noticed the use of `mesh.cell_centers().points` to get the center coordinates of the cell grid, which is used to calculate the distance between two adjacent grid cells. But for TOUGH it should use the vertices of the pairwise Delaunay triangles of the voronoi diagram as the "centers" of the grid cells？

**Comment by jtecklenburg** [comment]
Sorry, I don't understand this comment. Can you add a sketch, please?

**Comment by wangjx520** [comment]
I've noticed that you defined the calc_conne method in tough3/helper.py to be used for calculating the connected area and distances, etc of neighboring meshes. This implementation uses the mesh center coordinates obtained from mesh.cell_centers().point provided by PyVista. But as far as I know, the center coordinates of the MESH file used by Tough should be the intersection of the perpendicular bisectors of the sides of the Tyson polygon, which are the points in your source code that generate the Voronoi (Voronoi(points,boundary=boundary)). 
![image](https://github.com/user-attachments/assets/d83bb9cc-8d81-424f-a259-9f87fce2f55c)
The vertices of those Delaunay triangles.

**Reply by jtecklenburg** [reply]
I think I understand the problem. In your mesh above, you have two types of elements. There is one inner element and a lot of boundary elements.

In a more or less realistic reservoir model with co2 injection (This is my problem), there are a lot of inner elements and one or more inner elements with a source term for CO2 injection. The boundary elements are either no flow boundaries or Dirichlet boundaries with a a fixed pressure, saturation and so on. In order to avoid boundary effects, the distance between the source element and the boundary elements needs to large. 

Now I hope, that the edges of the Delaunay triangulation that is dual to the Voronoi diagram matches the points in mesh.cell_centers().point for the inner elements. I need to check this.

The problem you mentioned may apply to the boundary elements, only. For my models, this not a problem, because I try to avoid boundary effects and I mitigate this problem by defining a suitable buffer size, when creating the Voronoi diagram.
`voro = Voronoi(points[:, :2], buffer_size=40.0)`

What is your application you want to solve using Voronoi meshes? Why may this be a problem in your application?

**Comment by wangjx520** [comment]
And, I think the 
`elif dim.upper() == "Y" or dim.upper() == "XZ":`
`    return np.hstack((points[:, 0], z, points[:, 1]))`
may need to be changed to  
`elif dim.upper() == "Y" or dim.upper() == "XZ":`
`    return np.hstack((points[:, 0:1], z, points[:, 1:2]))`
